File: Server/Members/Member.php
<?php
require_once($_SERVER["DOCUMENT_ROOT"].'/Shared/Encodable.php');
class Member implements Encodable
{
	private $login;
	private $favorites;
	private $history;
	public function __construct($login,$favorite=array(),$history=array()){
		$this->login=$login;
		$this->favorites=$favorite;
		$this->history=$history;
	}
	
	public function getLogin(){
		return $this->login;
	}
	
	public function getFavorites(){
		return $this->favorites;
	}	
	
	public function getHistory(){
		return $this->history;
	}
	
	public function setFavorites($favorites){
		$this->favorites=$favorites;
	}
	
	public function setHistory($history){
		$this->history=$history;
	}
	
	public function encode(){
		$json = '{"login":"'.$this->login.'","fnb":"'.count($this->favorites).'","hnb":"'.count($this->history).'","favorites":[';
		foreach($this->favorites as $i=>$fav){
			$json.=$fav->encode();
			if($i>0)
				$json.=',';
		}
		$json.='],"history":[';
		foreach($this->history as $i=>$hist){
			$json.=$hist->encode();
			if($i>0)
				$json.=',';
		}
		$json.=']}';
		return $json;
	}
}
?>

File: Client/src/pcd/telecomnancy/model/member/Member.java
package pcd.telecomnancy.model.member;

import java.util.ArrayList;
import java.util.List;

import pcd.telecomnancy.model.routes.Route;




public class Member {
	private String login;
	private List<Route> history;
	private List<Favorite> favorites;
	
	public Member(){
		this.setLogin("Invité");
		history=new ArrayList<Route>();
		favorites=new ArrayList<Favorite>();
	}
	
	public Member(String login){
		this();
		this.setLogin(login);
	}

	public List<Favorite> getFavorites() {
		return favorites;
	}

	public void setFavorites(List<Favorite> favorites) {
		this.favorites = favorites;
	}

	public List<Route> getHistory() {
		return history;
	}

	public void setHistory(List<Route> history) {
		this.history = history;
	}

	public String getLogin() {
		return login;
	}

	public void setLogin(String login) {
		this.login = login;
	}
	
	public boolean isGuest(){
		return login.equals("Invité");
	}
}
